src/content.rs

```diff
@@ -407,6 +504,40 @@ impl Array {
         }
     }
 
+    /// Returns an iterator over the array items. Items must be dereferenced to access
+    /// the `Value`. For example:
+    ///
+    /// ```
+    /// use json5format::*;
+    /// let parsed_document = ParsedDocument::from_str("{}", None)?;
+    /// for item in parsed_document.content.items() {
+    ///     assert!(!(*item).is_primitive());
+    /// }
+    /// # Ok::<(),anyhow::Error>(())
+    /// ```
+    pub fn items(&self) -> impl Iterator<Item = Ref<'_, Value>> {
+        self.items.iter().map(|rc| rc.borrow())
+    }
+
+    /// As in `Array::items`, returns an iterator over the array items, but with mutable references.
+    #[inline]
+    pub fn items_mut(&mut self) -> impl Iterator<Item = RefMut<'_, Value>> {
```

Comment:
All of the new functions should have tests, especially since they are public.

You could do this via doc tests, which would really be beneficial for API users.

for example, for trailing_comments, something like this would probably work:

```rustdoc
    /// ```
    /// use json5format::*;
    /// let json5=r##"{
    /// }
    /// // trailing line comment at end of document
    /// "##;
    /// let parsed_document = ParsedDocument::from_str(json5, None)?;
    /// match parsed_document.content.trailing_comments()[0] {
    ///     Comment::Line(s) => assert_eq!(s, " trailing line comment at end of document"),
    ///     _ => panic!("not a Comment::Line"),
    /// }
    /// # Ok::<(),anyhow::Error>(())
    /// ```
```